app/controllers/providers/allocations_controller.rb

```diff
@@ -103,3 +103,138 @@ def requested?
+      {
+        training_providers: training_providers_from_query_without_associated,
+      }
+    else
+      {
+        training_providers: training_providers_without_associated,
+        form_object: form_object,
+      }
+    end
+  end
+
+private
+
+  def form_object
+    @form_object ||= InitialRequestForm.new
+  end
+
+  def allocations
+    @allocations ||= Allocation.includes(:provider, :accredited_body)
+                               .where(provider_code: provider.provider_code)
+                               .all
+  end
+
+  def training_providers_with_fee_paying_pe_course
+    @training_providers_with_fee_paying_pe_course ||= provider.training_providers(
+      recruitment_cycle_year: recruitment_cycle.year,
+      "filter[subjects]": PE_SUBJECT_CODE,
+      "filter[funding_type]": "fee",
+    )
+  end
+
+  def all_training_providers
+    @all_training_providers ||=
+      provider.training_providers(
+        recruitment_cycle_year: recruitment_cycle.year,
+      )
+  end
+
+  def training_providers_with_previous_allocations
+    @training_providers_with_previous_allocations ||= allocations.map(&:provider)
+  end
+
+  def associated_training_providers
+    training_providers_with_previous_allocations + training_providers_with_fee_paying_pe_course
+  end
+
+  def training_providers_without_associated
+    return @training_providers_without_associated if @training_providers_without_associated
+
+    ids_to_reject = associated_training_providers.map(&:id)
+
+    @training_providers_without_associated = all_training_providers.reject do |provider|
+      ids_to_reject.include?(provider.id)
+    end
+  end
+
+  def training_providers_from_query
+    return @training_providers_from_query if @training_providers_from_query
+
+    query = params[:training_provider_query]
+    @training_providers_from_query ||= ProviderSuggestion.suggest(query)
+  end
+
+  def training_providers_from_query_without_associated
+    ids_to_reject = associated_training_providers.map(&:id)
+
+    training_providers_from_query.reject do |r|
+      ids_to_reject.include?(r.id)
+    end
+  end
+
+  def recruitment_cycle
+    return @recruitment_cycle if @recruitment_cycle
+
+    cycle_year = params.fetch(:year, Settings.current_cycle)
+
+    @recruitment_cycle = RecruitmentCycle.find(cycle_year).first
+  end
+
+  def provider
+    @provider ||= Provider
+      .where(recruitment_cycle_year: recruitment_cycle.year)
+      .find(params[:provider_code])
+      .first
+  end
+
+  def empty_search_results?
+    return @empty_search_results if @empty_search_results
+
+    @empty_search_results = params[:training_provider_code].blank? && params[:training_provider_query].present? && training_providers_from_query_without_associated.empty?
+
+    form_object.add_no_results_error if @empty_search_results
+
+    @empty_search_results
+  end
+
+  def pick_a_provider_page?
+    params[:training_provider_code].blank? && params[:training_provider_query].present?
```

Comment:
What is the benefit of adding `.present?` here and for other predicate methods? 
Won't `params[:training_provider_query]` work in the same way?